### ISSUE TITLE
 Warning: grad and param do not obey the gradient layout contract.

### ISSUE THREAD

**Issue by mohaEs** [lucidrains/byol-pytorch#74]
Do you guys get a similar warning when using it?

Warning: grad and param do not obey the gradient layout contract. This is not an error, but may impair performance.
grad.sizes() = [512, 256, 1, 1], strides() = [256, 1, 1, 1]
param.sizes() = [512, 256, 1, 1], strides() = [256, 1, 256, 256] (function operator())

**Comment by hcmea** [comment]
I just started training BYOL with my custom dataset and got the same warning too. Did your model performed better on the downstream task? Does this warning effect the performance?

**Comment by zmzhang2000** [comment]
I got the same warning too. And the loss becomes NaN during training. Still not figured it out.

**Comment by zmzhang2000** [comment]
> I got the same warning too. And the loss becomes NaN during training. Still not figured it out.

I set the input to be contiguous  and then the warning disappeared.

`input = input.contiguous()`
